### PR TITLE
controller/backup-operator: support s3Path for BackupCRStatus

### PR DESCRIPTION
**Pull request by fanminshi**

### CODE REVIEW

**Comment by hongchaodeng** [comment]
Can you manually try it out and print the result (status) here?

**Comment by fanminshi** [comment]
@hongchaodeng I can do that. But I was hoping to write test with along this pr when https://github.com/coreos/etcd-operator/pull/1497 gets merged.

**Comment by hongchaodeng** [comment]
Let's finish this before worrying about the test. Finish tasks one by one.

**Comment by fanminshi** [comment]
manual test:

```
$ kubectl get EtcdBackup example-etcd-cluster -o yaml
apiVersion: etcd.database.coreos.com/v1beta2
kind: EtcdBackup
metadata:
  clusterName: ""
  creationTimestamp: 2017-10-27T00:07:11Z
  generation: 0
  name: example-etcd-cluster
  namespace: default
  resourceVersion: "148495"
  selfLink: /apis/etcd.database.coreos.com/v1beta2/namespaces/default/etcdbackups/example-etcd-cluster
  uid: c7e0b71b-baaa-11e7-ba7b-080027f6acec
spec:
  clusterName: example-etcd-cluster
  s3:
    awsSecret: aws
    s3Bucket: jenkins-etcd-operator
  storageType: S3
status:
  s3Path: jenkins-etcd-operator/v1/default/example-etcd-cluster/3.1.8_0000000000000001_etcd.backup
  succeeded: true
```
verify s3 path:
```
aws s3 ls jenkins-etcd-operator/v1/default/example-etcd-cluster/3.1.8_0000000000000001_etcd.backup
2017-10-26 17:07:13      24608 3.1.8_0000000000000001_etcd.backup
```

File: example/etcd-backup-operator/backup_cr.yaml [hunk 1]
```diff
@@ -6,5 +6,5 @@ spec:
   clusterName: example-etcd-cluster
   storageType: S3
   s3:
-    s3Bucket: <s3-bucket-name>
-    awsSecret: <aws-secret>
+    s3Bucket: jenkins-etcd-operator
```
Comment (hongchaodeng) [inline example/etcd-backup-operator/backup_cr.yaml]: Don't change this

File: pkg/controller/backup-operator/s3_backup.go [hunk 2]
```diff
@@ -16,28 +16,39 @@ package controller
...
+	"github.com/coreos/etcd-operator/pkg/backup/util"
 	"github.com/coreos/etcd-operator/pkg/controller/controllerutil"
 
 	"k8s.io/client-go/kubernetes"
 )
 
 // TODO: replace this with generic backend interface for other options (PV, Azure)
 // handleS3 backups up etcd cluster to s3.
-func handleS3(kubecli kubernetes.Interface, s3 *api.S3Source, namespace, clusterName string) error {
+func handleS3(kubecli kubernetes.Interface, s3 *api.S3Source, namespace, clusterName string) (string, error) {
```
Comment (hongchaodeng) [inline pkg/controller/backup-operator/s3_backup.go]: doc about what it returns.

**Comment by fanminshi** [comment]
PTAL cc/ @hongchaodeng

File: pkg/controller/backup-operator/s3_backup.go (hunk 2 above)
Comment (fanminshi) [reply pkg/controller/backup-operator/s3_backup.go]: done.

File: pkg/controller/backup-operator/sync.go [hunk 3]
```diff
@@ -109,12 +110,23 @@ func (b *Backup) handleErr(err error, key interface{}) {
 	b.logger.Infof("Dropping etcd backup (%v) out of the queue: %v", key, err)
 }
 
-func (b *Backup) handleBackup(spec *api.BackupSpec) error {
+type backupStatus struct {
```
Comment (hongchaodeng) [inline pkg/controller/backup-operator/sync.go]: Don't redefine this.
Just reuse api.BackupCRStatus, ignore the succeeded and Reason fields

Comment (fanminshi) [reply pkg/controller/backup-operator/sync.go]: done.!

**Comment by hongchaodeng** [comment]
lgtm

**Comment by fanminshi** [comment]
merge when green.

**Pull request merged**